astroquery/wfau/core.py

```diff
@@ -0,0 +1,903 @@
+        """
+        if self.session is None:
+            return False
+        for cookie in self.session.cookies:
+            if cookie.is_expired():
+                return False
+        return True
+
+    def _args_to_payload(self, *args, **kwargs):
+        request_payload = {}
+
+        request_payload['database'] = kwargs.get('database', self.database)
+
+        programme_id = kwargs.get('programme_id', self.programme_id)
+
+        request_payload['programmeID'] = self._verify_programme_id(
+            programme_id, query_type=kwargs['query_type'])
+        sys = self._parse_system(kwargs.get('system'))
+        request_payload['sys'] = sys
+        if sys == 'J':
+            C = commons.parse_coordinates(args[0]).transform_to(coord.ICRS)
+            request_payload['ra'] = C.ra.degree
+            request_payload['dec'] = C.dec.degree
+        elif sys == 'G':
+            C = commons.parse_coordinates(args[0]).transform_to(coord.Galactic)
+            request_payload['ra'] = C.l.degree
+            request_payload['dec'] = C.b.degree
+        return request_payload
+
+    def _verify_programme_id(self, pid, query_type='catalog'):
+        """
+        Verify the programme ID is valid for the query being executed.
+
+        Parameters
+        ----------
+        pid : int or str
+            The programme ID, either an integer (i.e., the # that will get passed
+            to the URL) or a string using the three-letter acronym for the
+            programme or its long name
+
+        Returns
+        -------
+        pid : int
+            Returns the integer version of the programme ID
+
+        Raises
+        ------
+        ValueError if the pid is 'all' and the query type is a catalog.
+        You can query all surveys for images, but not all catalogs.
+        """
+        if pid == 'all' and query_type == 'image':
+            return 'all'
+        elif pid == 'all' and query_type == 'catalog':
+            raise ValueError(
+                "Cannot query all catalogs at once. Valid catalogs are: {0}.\n"
+                "Change programmeID to one of these."
+                .format(",".join(self.programmes_short.keys())))
+        elif pid in self.programmes_long:
+            return self.programmes_long[pid]
+        elif pid in self.programmes_short:
+            return self.programmes_short[pid]
+        elif query_type != 'image':
+            raise ValueError("programme_id {0} not recognized".format(pid))
+
+    def _parse_system(self, system):
+        if system is None:
+            return 'J'
+        elif system.lower() in ('g', 'gal', 'galactic'):
+            return 'G'
+        elif system.lower() in ('j', 'j2000', 'celestical', 'radec'):
+            return 'J'
+
+    def get_images(self, coordinates, waveband='all', frame_type='stack',
+                   image_width=1 * u.arcmin, image_height=None, radius=None,
+                   database=None, programme_id=None,
```

Comment:
super nitpicky, but both ``database`` and ``programme_id`` could be consistent here with the ones of ``__init__``. Either use None there, too or here ``''``

Reply:
`__init__` defines the overall default.  `None` is "use that default".  We only want the default specified in one place.

Reply:
and will those default show up in the docstring? I mean it can be quite confusing when these come up as `None` after checking `get_images?`, but actually the __init__ defaults are used.